plugins/pluginmanagers/lz-n.nix

```diff
@@ -27,7 +27,22 @@ let
         cmd = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on command.";
         ft = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on filetype.";
         colorscheme = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on colorscheme.";
-        keys = helpers.mkNullOrOption (listOf helpers.keymaps.mapOptionSubmodule) "Lazy-load on key mapping. Use the same format as `config.keymaps`.";
+        keys =
+          with helpers.nixvimTypes;
+          let
+            singleStrToLuaObj =
+              s: if hasInfix "{" s then helpers.mkRaw s else helpers.toLuaObject (helpers.mkRaw s);
+            keysType = coercedTo str singleStrToLuaObj strLua;
```

Comment:
Not sure coerced is appropriate here, unless you're using it for a particular reason?

If we're matching upstream's spec they either accept `str` or they don't. Either way we shouldn't need to do any coercion. The `either` or `oneOf` type should be ok if `str` is accepted.

Reply:
Ah, I see. You're using it because you want `strLua` **and** `str`, and you are detecting lua based on table syntax being used.

While clever, and helpful in most scenarios, I think we should use `maybeRaw str` here (equivalent to `either str rawLua`).

Unfortunately this will be affected by #1618, but that should be fixed separately.

1. Detecting `"{"` will fail if that char is used as part of an actual keymapping.
2. We're already debating amongst maintainers whether `strLua` is too magic to use at all
3. If there's any confusion with actual string types, we should not use `strLua`